src/journal/journald-context.c

```diff
@@ -550,25 +589,38 @@ void client_context_maybe_refresh(
 }
 
 static void client_context_try_shrink_to(Server *s, size_t limit) {
+        ClientContext *c;
+        usec_t t;
+
         assert(s);
 
+        /* Flush any cache entries for PIDs that have already moved on. Don't do this
```

Comment:
I wonder if we shouldn't do liveness check in a separate function triggered by the timer event from the event loop.